rules/prefer-global-this.js

```diff
@@ -0,0 +1,491 @@
+	[MESSAGE_ID_SUGGESTION]: 'Replace `{{value}}` with `{{replacement}}`.',
+};
+
+/**
+Find the variable in the scope.
+
+@param {import('eslint').Scope.Scope} scope
+@param {string} variableName
+*/
+function findVariableInScope(scope, variableName) {
+	if (!scope || scope.type === 'global') {
+		return;
+	}
+
+	const variable = scope.variables.find(variable => variable.name === variableName);
+
+	if (variable) {
+		return variable;
+	}
+
+	return findVariableInScope(scope.upper, variableName);
+}
+
+const globalIdentifier = new Set(['window', 'self', 'global']);
+
+const windowSpecificAPIs = new Set([
+	// Properties and methods
+	// https://html.spec.whatwg.org/multipage/nav-history-apis.html#the-window-object
+	'window',
+	'self',
+	'document',
+	'name',
+	'location',
+	'history',
+	'navigation',
+	'customElements',
+	'locationbar',
+	'menubar',
+	'personalbar',
+	'scrollbars',
+	'statusbar',
+	'toolbar',
+	'status',
+	'close',
+	'closed',
+	'stop',
+	'focus',
+	'blur',
+	'frames',
+	'length',
+	'top',
+	'opener',
+	'parent',
+	'frameElement',
+	'open',
+	'object',
+	'navigator',
+	'clientInformation',
+	'originAgentCluster',
+	'alert',
+	'confirm',
+	'prompt',
+	'print',
+	'postMessage',
+
+	// global events
+	'onabort',
+	'onauxclick',
+	'onbeforeinput',
+	'onbeforematch',
+	'onbeforetoggle',
+	'onblur',
+	'oncancel',
+	'oncanplay',
+	'oncanplaythrough',
+	'onchange',
+	'onclick',
+	'onclose',
+	'oncontextlost',
+	'oncontextmenu',
+	'oncontextrestored',
+	'oncopy',
+	'oncuechange',
+	'oncut',
+	'ondblclick',
+	'ondrag',
+	'ondragend',
+	'ondragenter',
+	'ondragleave',
+	'ondragover',
+	'ondragstart',
+	'ondrop',
+	'ondurationchange',
+	'onemptied',
+	'onended',
+	'onerror',
+	'onfocus',
+	'onformdata',
+	'oninput',
+	'oninvalid',
+	'onkeydown',
+	'onkeypress',
+	'onkeyup',
+	'onload',
+	'onloadeddata',
+	'onloadedmetadata',
+	'onloadstart',
+	'onmousedown',
+	'onmouseenter',
+	'onmouseleave',
+	'onmousemove',
+	'onmouseout',
+	'onmouseover',
+	'onmouseup',
+	'onpaste',
+	'onpause',
+	'onplay',
+	'onplaying',
+	'onprogress',
+	'onratechange',
+	'onreset',
+	'onresize',
+	'onscroll',
+	'onscrollend',
+	'onsecuritypolicyviolation',
+	'onseeked',
+	'onseeking',
+	'onselect',
+	'onslotchange',
+	'onstalled',
+	'onsubmit',
+	'onsuspend',
+	'ontimeupdate',
+	'ontoggle',
+	'onvolumechange',
+	'onwaiting',
+	'onwebkitanimationend',
+	'onwebkitanimationiteration',
+	'onwebkitanimationstart',
+	'onwebkittransitionend',
+	'onwheel',
+
+	// Window events
+	'onafterprint',
+	'onbeforeprint',
+	'onbeforeunload',
+	'onhashchange',
+	'onlanguagechange',
+	'onmessage',
+	'onmessageerror',
+	'onoffline',
+	'ononline',
+	'onpagehide',
+	'onpagereveal',
+	'onpageshow',
+	'onpageswap',
+	'onpopstate',
+	'onrejectionhandled',
+	'onstorage',
+	'onunhandledrejection',
+	'onunload',
+
+	// https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-flow
+	'addEventListener',
+	'removeEventListener',
+	'dispatchEvent',
```

Comment:
In a future PR, not now, it would be cool to warn on _some_ events.

```js
window.addEventListener('resize', fn); // ✅ 
window.addEventListener('play', fn); // ❌ 
globalThis.addEventListener('play', fn); // ✅ 
```

Reply:
Implemented and add test.